File: storage/data_transpiled/at_coder/agc031/B/4717018.java
public static void main ( String [ ] args ) {
  final int mod = 10 * 9 + 7 ;
  int n = Integer . parseInt ( input . nextLine ( ) ) ;
  int [ ] nums = new int [ n ] ;
  nums [ 0 ] = - 1 ;
  for ( int i = 0 ;
  i < n ;
  i ++ ) {
    int tmp = Integer . parseInt ( input . nextLine ( ) ) ;
    if ( nums [ i ] != tmp ) {
      nums [ i ] = tmp ;
    }
  }
  int m = nums . length ;
  int [ ] dp = new int [ m ] ;
  int [ ] colors = new int [ Math . max ( nums . length + 1 , m ) ] ;
  for ( int i = 1 ;
  i < m ;
  i ++ ) {
    dp [ i ] = 1 ;
    colors [ i ] = 0 ;
  }
  for ( int i = 1 ;
  i < m ;
  i ++ ) {
    dp [ i ] = ( dp [ i - 1 ] + colors [ nums [ i ] ] ) % mod ;
    colors [ nums [ i ] ] = dp [ i ] ;
  }
  System . out . println ( dp [ dp . length - 1 ] ) ;
}

File: storage/data_original/at_coder/agc031/B/4717018.py
import numpy as np

def main():
  mod = 10 ** 9 + 7

  n = int(input())
  nums = [-1]
  for _ in range(n):
      tmp = int(input())
      if nums[-1] != tmp:
          nums.append(tmp)
  #print(nums)

  m = len(nums)

  dp = [1 for _ in range(m)] # i?????????????i+1???????????????????
  colors = [0 for _ in range(max(nums) + 1)] #?????????i?????????

  for i in range(1, m):
    dp[i] = (dp[i - 1] + colors[nums[i]]) % mod #i????????????????????????i - 1????????????i???????????????
    colors[nums[i]] = dp[i]
    #print(dp, colors)


  print(dp[-1])
  

if __name__ == '__main__':
  main()